x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/dataframe/extractor/ExtractedFieldsDetectorTests.java

```diff
@@ -1028,7 +1028,12 @@ public void testDetect_GivenParentAndMultiFieldBothAggregatable() {
             fieldCapabilities,
             Collections.emptyMap()
         );
-        Tuple<ExtractedFields, List<FieldSelection>> fieldExtraction = extractedFieldsDetector.detect();
+        Tuple<ExtractedFields, List<FieldSelection>> fieldExtraction = extractedFieldsDetector.detect();// causing java.lang.IllegalArgumentException: cannot retrieve field [field_2] because it has no mappings
```

Comment:
This exception can be avoided by moving the new lines 223-225 in the other file down to line 231.